### PR TITLE
Mig/10.0

### PR DESCRIPTION
**Pull request by mstuttgart**
I will fix code conflict.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,8 @@
-[![Build Status](https://travis-ci.org/JayVora-SerpentCS/Jasperv8.svg?branch=master)](https://travis-ci.org/JayVora-SerpentCS/Jasperv8)
 [![Build Status](https://travis-ci.org/JayVora-SerpentCS/Jasperv8.svg?branch=8.0)](https://travis-ci.org/JayVora-SerpentCS/Jasperv8)
 ========
```
Comment (JayVora-SerpentCS) [inline README.md]: branch=10.0

File: jasper_reports/JasperReports/jasper_server.py [hunk 2]
```diff
@@ -60,6 +60,16 @@ def path(self):
         return os.path.abspath(os.path.dirname(__file__))
 
     def start(self):
+        java_path = self.javapath
+        if java_path is False:
+            raise UserError(_('Java Path Not Found !'),_('Please add java path into the jasper configuration page under the company form view'))
+        else :
```
Comment (JayVora-SerpentCS) [inline jasper_reports/JasperReports/jasper_server.py]: Would be so nice if you can follow pep-8.

File: jasper_reports/jasper_report.py [hunk 3]
```diff
@@ -185,11 +185,8 @@ def execute(self):
         logger.info("ELAPSED: %f" % elapsed)
 
         # Read data from the generated file and return it
-        f = open(output_file, 'rb')
-        try:
+        with open(output_file, 'rb') as f:
             data = f.read()
-        finally:
-            f.close()
```
Comment (JayVora-SerpentCS) [inline jasper_reports/jasper_report.py:191]: We should keep exceptions, isnt it?

File: jasper_reports/jasper_report.py [hunk 4]
```diff
@@ -265,11 +262,19 @@ def execute_report(self, data_file, output_file, sub_report_data_files):
...
         server = JasperServer(int(tools.config['jasperport']))
+        server.setPidFile(tools.config['jasperpid'])
+#        java path for jasper server
+        company_rec = self.env['res.users'].browse(self.uid).company_id
+        java_path = ''
+        if company_rec and company_rec.id:
+            java_path = company_rec.java_path
+            server.javapath = java_path
+        server.javapath = java_path
+
```
Comment (JayVora-SerpentCS) [inline jasper_reports/jasper_report.py]: do in one line:
server.javapath = company_rec and company_rec.java_path or ''

File: jasper_reports/wizard/jasper_create_data_template.py [hunk 5]
```diff
@@ -36,6 +36,10 @@
 
 
 class CreateDataTemplate(models.TransientModel):
+<<<<<<< Updated upstream
+=======
+
+>>>>>>> Stashed changes
     _name = 'jasper.create.data.template'
```
Comment (JayVora-SerpentCS) [inline jasper_reports/wizard/jasper_create_data_template.py:40]: conflict my friend!

File: jasper_reports/jasper_report.py (hunk 3 above)
Comment (mstuttgart) [reply jasper_reports/jasper_report.py:191]: It is not need. This *with* command execute *f.close()* in background.

File: jasper_reports/JasperReports/record_data_generator.py [hunk 6]
```diff
@@ -111,5 +176,11 @@ def generate(self, fileName):
...
             f.close()
+>>>>>>> upstream/10.0:jasper_reports/JasperReports/RecordDataGenerator.py
 
-# vim:noexpandtab:smartindent:tabstop=8:softtabstop=8:shiftwidth=8:
+                value_node = self.document.createTextNode(value)
+                field_node.appendChild(value_node)
+
+        # Once created, the only missing step is to store the XML into a file
+        with codecs.open(file_name, 'wb+', 'utf-8') as f:
+            top_node.writexml(f)
```
Comment (JayVora-SerpentCS) [inline jasper_reports/JasperReports/record_data_generator.py:125]: Man, so many conflicts!

Comment (mstuttgart) [reply jasper_reports/JasperReports/record_data_generator.py:125]: I need the rebase my PR. It is work in progress.

File: jasper_reports/JasperReports/jasper_server.py (hunk 2 above)
Comment (mstuttgart) [reply jasper_reports/JasperReports/jasper_server.py]: Fixed

File: jasper_reports/jasper_report.py (hunk 4 above)
Comment (mstuttgart) [reply jasper_reports/jasper_report.py]: Fixed

File: jasper_reports/wizard/jasper_create_data_template.py (hunk 5 above)
Comment (mstuttgart) [reply jasper_reports/wizard/jasper_create_data_template.py:40]: Fixed

File: jasper_reports/JasperReports/record_data_generator.py (hunk 6 above)
Comment (mstuttgart) [reply jasper_reports/JasperReports/record_data_generator.py:125]: Fixed

**Comment by mstuttgart** [comment]
Travis fixed: https://travis-ci.org/multidadosti-erp/Jasperv8

**Comment by JayVora-SerpentCS** [comment]
Would be great if you can fix travis

**Comment by mstuttgart** [comment]
This travis is "travis-ci/pr". He must be fixed in "10.0" branch, not in this PR.

File: jasper_reports/__manifest__.py [hunk 7]
```diff
@@ -59,7 +59,7 @@
         'data/jasper_data.xml',
         'res_company_view.xml',
     ],
-    "active": False,
+    "active": True,
     "installable": True,
```
Comment (JayVora-SerpentCS) [inline jasper_reports/__manifest__.py:63]: make it auto_install

File: jasper_reports/JasperReports/abstract_data_generator.py [hunk 8]
```diff
@@ -31,6 +31,10 @@
 
 
 class AbstractDataGenerator:
+
+    def __init__(self):
+        pass
+
     # Simple function all DataGenerators should implement
-    def generate(self, fileName):
+    def generate(self, filename):
```
Comment (JayVora-SerpentCS) [inline jasper_reports/JasperReports/abstract_data_generator.py:39]: its a declaration, nowhere it is used?

File: jasper_reports/JasperReports/browse_data_generator.py [hunk 9]
```diff
@@ -67,175 +68,170 @@ def warning(self, message):
...
+            new_records = self.generate_ids(record, relations, '',
+                                            [{'root': record}])
             copies = 1
-            if(self.report.copiesField() and record.__hasattr__
-               (self.report.copiesField())):
-                copies = int(record.__getattr__(self.report.copiesField()))
-            for new in newRecords:
+            if self.report.copiesField() and \
+                    record.__hasattr__(self.report.copies_field):
+                copies = int(record.__getattr__(self.report.copies_field))
```
Comment (JayVora-SerpentCS) [inline jasper_reports/JasperReports/browse_data_generator.py:194]: pep-8 fix plz

File: jasper_reports/__manifest__.py [hunk 10]
```diff
@@ -40,20 +40,26 @@
...
+    "data": [
+        'wizard/jasper_create_data_template.xml',
+        'jasper_wizard.xml',
+        'report_xml_view.xml',
+        'security/ir.model.access.csv',
+        'data/jasper_data.xml',
+        'res_company_view.xml',
+    ],
+    "active": True,
     "installable": True,
```
Comment (JayVora-SerpentCS) [inline jasper_reports/__manifest__.py:63]: auto_install:True

File: jasper_reports/JasperReports/browse_data_generator.py (hunk 9 above)
Comment (mstuttgart) [reply jasper_reports/JasperReports/browse_data_generator.py:194]: This code pass in pep8 check: https://travis-ci.org/multidadosti-erp/Jasperv8/jobs/188831847

File: jasper_reports/JasperReports/abstract_data_generator.py (hunk 8 above)
Comment (mstuttgart) [reply jasper_reports/JasperReports/abstract_data_generator.py:39]: It is override in subclasses. 
I not add this code, only migrate to Odoo 10.0 and adjust PEP8 in python code

**Comment by JayVora-SerpentCS** [comment]
I would appreciate if you can add up 2 sample reports. We should also work on idex.html

**Pull request merged**